shop/cmd/main.go

```diff
@@ -20,6 +33,29 @@ func main() {
 	if err != nil {
 		logger.Fatal("Failed to initialize PostgreSQL connection: " + err.Error())
 	}
```

Comment:
handle graceful shutdown for postgres

see defer statement for pg in auth service

Reply:
also rename the variable from postgres to something else as it collides with the package name